ocrd_tesserocr/common.py

```diff
@@ -5,43 +5,145 @@
+    if isinstance(coordinates, np.ndarray):
+        coordinates = list(map(tuple, coordinates))
     ImageDraw.Draw(mask).polygon(coordinates, outline=1, fill=255)
     return mask
 
 # to be refactored into core (as function in ocrd_utils):
-def rotate_polygon(coordinates, angle, orig={'x': 0, 'y': 0}):
-    # if the region image has been rotated, we must also
-    # rotate the coordinates of the line
-    # (which relate to the top page image)
-    # in the same direction but with inverse transformation
-    # matrix (i.e. passive rotation), and
-    # (since the region was rotated around its center,
-    #  but our coordinates are now relative to the top left)
-    # by first translating to center of region, then
-    # rotating around that center, and translating back:
-    # point := (point - region_center) * region_rotation + region_center
-    # moreover, since rotation has reshaped/expanded the image,
-    # the line coordinates must be offset by those additional pixels:
-    # point := point + 0.5 * (new_region_size - old_region_size)
+def image_from_polygon(image, polygon):
+    """"Mask an image with a polygon.
+    
+    Given a PIL.Image `image` and a numpy array `polygon`
+    of relative coordinates into the image, fill everything
+    outside the polygon hull to the background. Since `image`
+    is not necessarily binarized yet, determine the background
+    from the median color (instead of white).
+    
+    Return a new PIL.Image.
+    """
+    mask = polygon_mask(image, polygon)
+    # create a background image from its median color
+    # (in case it has not been binarized yet):
+    # array = np.asarray(image)
+    # background = np.median(array, axis=[0, 1], keepdims=True)
+    # array = np.broadcast_to(background.astype(np.uint8), array.shape)
+    background = ImageStat.Stat(image).median[0]
+    new_image = Image.new('L', image.size, background)
+    new_image.paste(image, mask=mask)
+    return new_image
+
+# to be refactored into core (as function in ocrd_utils):
+def crop_image(image, box=None):
+    """"Crop an image to a rectangle, filling with background.
+    
+    Given a PIL.Image `image` and a list `box` of the bounding
+    rectangle relative to the image, crop at the box coordinates,
+    filling everything outside `image` with the background.
+    (This covers the case where `box` indexes are negative or
+    larger than `image` width/height. PIL.Image.crop would fill
+    with black.) Since `image` is not necessarily binarized yet,
+    determine the background from the median color (instead of
+    white).
+    
+    Return a new PIL.Image.
+    """
+    # todo: perhaps we should issue a warning if we encounter this
+    # (It should be invalid in PAGE-XML to extend beyond parents.)
+    if not box:
+        box = (0, 0, image.width, image.height)
+    xywh = xywh_from_bbox(*box)
+    background = ImageStat.Stat(image).median[0]
+    new_image = Image.new(image.mode, (xywh['w'], xywh['h']),
```

Comment:
Is the median **defined** as white for binarized images?

Reply:
Yes, the median of binarized images is almost always white. In fact, Ocropy uses this to determine whether the input image was inverted (by counting the pixels above vs below median). The only case where it does not work would be very tiny regions (say a page number, cropped too tightly).

Reply:
@wrznr just had another idea: an option would be to use `RGBA` mode (with transparency) in PIL objects throughout...